### PR TITLE
Commit field to deploy

### PR DESCRIPTION
**Pull request by AliRizvi1**
https://github.com/dimagi/commcare-hq/issues/27839
Main feature of the above CEP.

Also related to https://github.com/dimagi/commcare-hq/pull/28202, where deploy history report was moved to a standalone page.

##### SUMMARY
Added a commit field to the HqDeploy model, enabling deploys to store the current commit directly. Similarly, `record_deploy_success` now has a `--commit` flag to pass the commit field when instantiating a new HqDeploy.

##### PRODUCT DESCRIPTION
Users will now see a commit field in the standalone Deploy History report, and will be able to pass in a commit field when using the management command `record_deploy_success`.

### CODE REVIEW

File: corehq/apps/hqadmin/migrations/0017_hqdeploy_commit.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2020-07-21 13:42
+from __future__ import unicode_literals
+
```
Comment (esoergel) [inline corehq/apps/hqadmin/migrations/0017_hqdeploy_commit.py:4]:
```suggestion
```
(These are only relevant to Python 2)

File: corehq/apps/hqadmin/reports.py [hunk 2]
```diff
@@ -317,3 +319,8 @@ def _format_date(self, date):
 
     def _hyperlink_diff_url(self, diff_url):
         return f'<a href="{diff_url}">Diff with previous</a>'
+
+    def _shorten_and_hyperlink_commit(self, commit_sha):
+        if commit_sha:
+            return f'<a href="https://github.com/dimagi/commcare-hq/commit/{commit_sha}">{commit_sha[:7]}</a>'
+        return None
```
Comment (esoergel) [inline corehq/apps/hqadmin/reports.py:326]: I don't think it's necessary to shorten the commit here - you can still link to the full thing.

Comment (AliRizvi1) [reply corehq/apps/hqadmin/reports.py:326]: image

This is what the shortened commit sha looks like. The hyperlink still takes you to the address with the full commit sha, but the shortening just looks cleaner in the table.

Comment (esoergel) [reply corehq/apps/hqadmin/reports.py:326]: Oh you're right, I misread that - your way is better.

**Review by esoergel — APPROVED** [review]

**Pull request merged**